### ISSUE TITLE
Error on image build but error log is not complete

### ISSUE THREAD

**Issue by gioppoluca** [CloudOpting/cloudopting-crane#7]
I'm trying to build an image using a working set of files (it builds locally on my machine).

I try to go to the API and get the info of the build but the log is truncated and cannot understand what the error is.

**Comment by j-fuentes** [comment]
Are you retrieving the error log with the API or inspecting the file?

**Comment by gioppoluca** [comment]
I use the /builder/images/{tk}/detail
API
And in the log field the info is truncated

**Comment by gioppoluca** [comment]
it just says build finished unsuccessfully in the description

**Comment by j-fuentes** [comment]
Yes, that field is limited.

Because of that, I recommend for debug purposes to [read from the log file](https://github.com/CloudOpting/cloudopting-crane/blob/master/docs/developwithcrane.md#extra-debug-errors).

But I am not sure if the file is overwritted with `build finished unsuccessfully`. Could you check it?

**Comment by gioppoluca** [comment]
mmm I see that you rename the puppet manifest to a "manifest.pp" but in the dockerfile I generate I have the generated name (webdk.pp).
Did we decide to change the name of the manifest inside the dockerfile?
If yes what is the expected content? path included

**Comment by gioppoluca** [comment]
can I advice you to leave the name of the puppet manifest as it is so the generated content will be good both when testing using crane and by hand (if you rename the dockerfile used by hand will not find the manifest.pp and I cannot have multiple manifest.pp in the same folder and I use a single folder for the generation)

**Comment by j-fuentes** [comment]
We decided to use standard names.

The Dockerfile can read the manifest.pp file from `images/<name of the image>/manifest.pp`.

But yes, I can left the name as is.

**Comment by gioppoluca** [comment]
also for me the webdk.pp manifest is in the root of the build context and not into a images/changed_generated_name folder
If you keef files as sent and use the -f flag for the build you can leave all in the context root and execute the builds without any problem keeping just the logs and all that things in the separated folders.
Leveraging on the -f flag could help

**Comment by gioppoluca** [comment]
The fact is that I have no knowledge of the crane environment and with the generated files as they are I need to be able to test both using crane AND by-hand.

**Comment by gioppoluca** [comment]
BTW we really need to be able to send the entire log back to the orchestrator or the service provider will not be albe to understand the error and he will not be able to access to the file system of the orchestrator

**Comment by j-fuentes** [comment]
yes, but do you want the entire log error in a http response? It could be a big big response.

**Comment by j-fuentes** [comment]
Should be fixed now (https://github.com/jmcerpa/cloudopting-crane/commit/25e8fe21a83fbe73fff37bd2f7236651366a81a6).

**Comment by gioppoluca** [comment]
Verify and close

**Comment by jmcerpa** [comment]
In the last dockService I add methods that allow managers to receive logs from crane. Manager can receive logs about image build and contexts, for example.
These methods are contextDetails, ImageDetails ...
Manager will receive a JSON, if you read the key "log" and "error_log" you can see what is happening in crane.
In this way, manager has more control over the process.
The old methods (getBuildInfo, getContexInfo...) receive a description, so you could not read the logs.

**Comment by jmcerpa** [comment]
When the manager receives these logs stored in elasticsearch?
